rpcserver.go

```diff
@@ -1174,9 +1178,56 @@ out:
 	return nil
 }
 
-// fetchActiveChannel attempts to locate a channel identified by its channel
+// AbandonChannel removes all channel state from the database except for a
+// close summary. This method can be used to get rid of permanently unusable
+// channels dued to bugs fixed in newer versions of lnd. 
+func (r *rpcServer) AbandonChannel(ctx context.Context,
+	in *lnrpc.AbandonChannelRequest) (*lnrpc.AbandonChannelResponse, error) {
+
+	if !DebugBuild {
+		return nil, fmt.Errorf("AbandonChannel RPC call only " +
+			"available in debug builds")
+	}
+
+	index := in.ChannelPoint.OutputIndex
+	txidHash, err := getChanPointFundingTxid(in.GetChannelPoint())
+	if err != nil {
+		return nil, err
+	}
+	txid, err := chainhash.NewHash(txidHash)
+	if err != nil {
+		return nil, err
+	}
+	chanPoint := wire.NewOutPoint(txid, index)
+
+	dbChan, err := r.fetchOpenDbChannel(*chanPoint)
```

Comment:
With ensuring that they're inactive, do you mean pending close? I am not sure if that is what you want. My impression is that abandon should remove the channel without trying to do anything else that might fail.

Reply:
In many cases, the users are also unable to actually fully read the borked channel from disk fully. As a result, we should check if we're unable to read the channel state from disk, and fall back to a more "aggressive" purge if needed. By more aggressive, I mean just deleting the bucket hierarchy that the channel resides in.

Reply:
But in that case, does lnd start up at all? I see that `fetchNodeChannels` is called on startup and deserializes the open channels. If it doesn't start, there is no opportunity to make the AbandonChannel call.

Isn't a database integrity check/correct on startup a better solution for this issue?

Reply:
Yes it's the case that `lnd` shouldn't actually get into this state, but it's a bit of a work-around to allow users that have channels that are _already_ borked (so can't properly deserialize for w/e reason) to cleanly remove the channel from the database.

Reply:
I see that those channels cannot be deserialized and that it shouldn't happen, but what is the best way to solve the problem? Extending the logic of `AbandonChannel` is not going to work because `lnd` isn't even starting and cannot accept the RPC call.

Reply:
So how it would work if we follow the line of this PR, is that when LND crashes on startup:
- Add continue/returns to make it start
- Use AbandonChannel to remove the borked channels from the database. In this process hope that the continue/returns are not placed in such a way that AbandonChannels isn't working anymore
- Revert code to original

This also means that quite some db code will be introduced to deal with this special situation.

I think I'd prefer a command line switch like `--fixdb` that executes a set of integrity checks and fixes problems. After making a backup of `channel.db`.

Reply:
I think the strategy you outlined in the bullet points makes sense, as this _should_ only affect a small set of users, and if they can create a debug build they can also add the necessary continues/returns.

Reply:
Looking at the changes required for AbandonChannel to be able to remove a borked channel, we'll be adding quite some code to channeldb just to be able to do that.